### ISSUE TITLE
Compiler Check fails for gcc 6.x

### ISSUE THREAD

**Issue by aichhorn** [TRIQS/triqs#382]
At the time of writing the compiler check, it was obviously not intended for version numbers larger than 5. The RegEx in FindCompilerCheck.cmake has only [2-5] and not [2-6] in line 24. Should be changed, otherwise the check fails.

**Comment by tayral** [comment]
Hi Markus,
Are you referring to FindCompilerCheck.cmake in the cmake folder? At line 24 (branch master or dev), I am seeing
```
MESSAGE(STATUS "Using clang version ${compiler_version}")
```
Under Ubuntu 16.04 (with branch 'dev', but the check is the same as on 'master'), the detection works:
```
$ cmake -DCMAKE_INSTALL_PREFIX=/blabla ../../source/triqs
-- Installation directory is /blabla
-- The C compiler identification is GNU 5.4.1
-- The CXX compiler identification is GNU 6.2.0
-- Check for working C compiler: /usr/bin/cc
-- Check for working C compiler: /usr/bin/cc -- works
...
```
Or did I misunderstand your message?
Thomas

**Comment by aichhorn** [comment]
Sorry, it is line 24 in the 1.3 version... It is the line
string(REGEX REPLACE ".*([2-5]\\.[0-9]\\.[0-9]).*" "\\1" compiler_version ${_compiler_output})
You see, the first number is limited to versions up to 5. If you change your gcc compilers ALL to 6.something, you will see the problem. I guess it works for you because gcc is still 5.4.1 in your case.
I changed the line to
string(REGEX REPLACE ".*([2-6]\\.[0-9]\\.[0-9]).*" "\\1" compiler_version ${_compiler_output})
and then it works of course also for new gcc versions.

**Comment by krivenko** [comment]
Hi Markus,

You must be referring to issue #328, which was fixed in May 2016, half a year after 1.3 release.

**Comment by aichhorn** [comment]
Okay, good. So for the master it is okay. I close this stuff.
